pinot-common/src/main/java/org/apache/pinot/common/utils/SegmentUtils.java

```diff
@@ -49,6 +49,24 @@ public static Integer getRealtimeSegmentPartitionId(String segmentName, String r
         ZKMetadataProvider.getSegmentZKMetadata(helixManager.getHelixPropertyStore(), realtimeTableName, segmentName);
     Preconditions.checkState(segmentZKMetadata != null,
         "Failed to find segment ZK metadata for segment: %s of table: %s", segmentName, realtimeTableName);
+    return getRealtimeSegmentPartitionId(segmentZKMetadata, partitionColumn);
+  }
+
+  @Nullable
+  public static Integer getRealtimeSegmentPartitionId(String segmentName, SegmentZKMetadata segmentZKMetadata,
+      @Nullable String partitionColumn) {
+    // A fast path if the segmentName is an LLC segment name: get the partition id from the name directly
+    LLCSegmentName llcSegmentName = LLCSegmentName.of(segmentName);
+    if (llcSegmentName != null) {
+      return llcSegmentName.getPartitionGroupId();
+    }
+    // Otherwise, retrieve the partition id from the segment zk metadata.
+    return getRealtimeSegmentPartitionId(segmentZKMetadata, partitionColumn);
+  }
+
+  @Nullable
+  public static Integer getRealtimeSegmentPartitionId(SegmentZKMetadata segmentZKMetadata,
```

Comment:
(minor) Make this private? We probably don't want user to directly use this method?